ksql-common/src/main/java/io/confluent/ksql/util/SchemaUtil.java

```diff
@@ -331,6 +331,42 @@ public static boolean isNumber(final Schema.Type type) {
     return ARITHMETIC_TYPES.contains(type);
   }
 
+  public static Schema ensureOptional(final Schema schema) {
```

Comment:
based on the first if branch, this method seems to only ensure that the _top level_ schema is optional, but based on the contents it will make the recursive structure optional... I think we should clarify this behavior and make it consistent.

For this PR, does it need to be recursively optional? The check in `KsqlSchema` suggests that only top-level matters.

Reply:
I think you've misread `KsqlSchema`.  It's _not_ requiring optional for the top level, but _does_ require optional for everything else.

However, I'm actually talking to @rodesai about this as I have a feeling we only require the first level of fields to be nullable, (to handle the situation where an error occurs calculating the column value and it is replaced by a `null`), not nested fields, key or value schemas.

Reply:
@rodesai has suggested that we just go with optional everywhere - as this is more consistent and allows for null fields in structs when we add ability to build structs from function calls.